### PR TITLE
Add OpenShift-related warning to addons documentation

### PR DESCRIPTION
**Pull request by jewertow**
Signed-off-by: Jacek Ewertowski <jewertow@redhat.com>

**Please provide a description of this PR:**

Look [here](https://github.com/jewertow/upstream-istio/tree/add-warning-for-openshift-monitoring/samples/addons#prometheus-operator) to see the result.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [X] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: samples/addons/README.md [hunk 1]
```diff
@@ -87,6 +87,13 @@ The [Prometheus Operator](https://github.com/coreos/prometheus-operator) manages
...
+> The example `PodMonitor` requires [metrics merging](https://istio.io/latest/docs/ops/integrations/prometheus/#option-1-metrics-merging) to be enabled. This is enabled by default.
 
-Note: The configurations here are only for Istio deployments, and do not scrape metrics from the Kubernetes components. See the [Cluster Monitoring](https://coreos.com/operators/prometheus/docs/latest/user-guides/cluster-monitoring.html) documentation for configuring this.
+> **Note**  
+> The configurations here are only for Istio deployments, and do not scrape metrics from the Kubernetes components.  
+> See the [Cluster Monitoring](https://coreos.com/operators/prometheus/docs/latest/user-guides/cluster-monitoring.html) documentation for configuring this.
+
+> **Warning**   
+> When the example `PodMonitor` is used with OpenShift Monitoring, it must be created in all namespaces where istio-proxies exist.  
+> This is because ignoring `namespaceSelector` for tenancy isolation.
```
[the remark this answers is not in the document]
Comment (jewertow) [reply samples/addons/README.md]: Thanks, done.

**Review by lei-tang — APPROVED** [review]

**Comment by ericvn** [comment]
This PR updated common files which is incorrect (unless the change is also needed common-files, as the next common-files update will override the file and we see the lint error again: https://github.com/istio/istio/pull/42804

**Comment by ericvn** [comment]
I'll include the changes, simply removing some spaces from the end of lines, in another PR.

**Comment by jewertow** [comment]
> I'll include the changes, simply removing some spaces from the end of lines, in another PR.

@ericvn those spaces at the end of lines were added on purpose to break lines.

**Comment by ericvn** [comment]
OK. Update made in my PR to keep them separated and still pass the existing lint tests.

**Pull request merged**